### DISCUSSION TITLE
Xcode General tab is missing

### DISCUSSION THREAD

**Discussion by asnagni** [axmolengine/axmol#1361] [Q&A] [answered]
- axmol version: 1.1
- devices test on: Simulator
- developing environments
   - NDK version: n/a
   - Xcode version: 14.3.1
   - Visual Studio: 
      - VS version: n/a
      - MSVC version: n/a
      - Windows SDK version: n/a
   - cmake version: 3.26.2
Steps to Reproduce:

Hi guys,
	First of all, I would to say congrats for the first official release 👍. We are trying to migrate from Cocos2d-x to Axmol. Unfortunately, after creating an Axmol project (for Xcode simulator) by following the steps provided on the website (https://github.com/axmolengine/axmol/blob/dev/docs/DevSetup.md), we noticed that from Xcode when you select your target, the General tab is missing. Not having the General Tab in the project makes it challenging. These are few challenges that we have:
  1) Select the orientation for iPhone independently
  2) Select a different orientation for iPad independently
  3) Integrating third party framework like “Firebase” or “onesignal” since they reference the General tab. 
  4) Configuring the Frameworks, Libraries, and Embedded Content
  5) etc...

Please let me know if we missed something in your tutorial/guide for creating an Axmol project.
If that was not the case, we would like to know if there is any plan to fix this issue in the near term? 

Thank you,
Stay safe

**Comment by rh101** [comment]
@asnagni ~Does it also happen when you create a iOS real device project?~  Just to confirm, is it this "General" tab that you are referring to:

![image](https://github.com/axmolengine/axmol/assets/8603230/5267a656-00c5-40ec-b802-61ecf91ed505)

That is from a project using Axmol (iOS arm64 target Xcode project).

Also, 3rd party frameworks, libraries and embedded content should be added via `CMakeLists.txt`, for your own sanity, because every time you add new files or content to your project that requires a re-run of cmake, it will re-create the Xcode project.  This means anything you added via the Xcode IDE will disappear.

**Comment by rh101** [comment]
I can see the problem too when using the following to create a project for the simulator:

`cmake -S . -B ./build-ios-sim -GXcode -DCMAKE_TOOLCHAIN_FILE=$AX_ROOT/cmake/ios.toolchain.cmake -DPLATFORM=SIMULATOR64`

![image](https://github.com/axmolengine/axmol/assets/8603230/e2ab21ba-de4a-453d-b058-6905cf179da1)

**Comment by asnagni** [comment]
Hi,
   1) We just created an iOS project targeting the device (am64) and the General tab is present. Thus we can confirm that this issue is just happening for the simulator projects.

   2) Thank you for the advice concerning the 3d party Framworks, Libraries, an Embedded Content, we do appreciate. 


Do you have any plan in the near future to fix the issue for  simulator project?

Thank you,
Stay safe

**Comment by rh101** [comment]
> Do you have any plan in the near future to fix the issue for simulator project?

That's a question for @halx99

**Comment by asnagni** [comment]
Ah ok I see.

@halx99 <https://github.com/halx99> is there any plan in the near future to fix this issue.


Thank you,
Stay safe

**Comment by halx99** [comment]
it's caused by explicit set `CMAKE_OSX_SYSROOT` to simulator https://github.com/axmolengine/axmol/blob/dev/1k/ios.cmake#L79C13-L79C30, only non-simualtor has `General` tab, even thourgh I use `xcodebuild -version -sdk iphonesimulator Path` query full path and set to `CMAKE_OSX_SYSROOT`, the `General` tab still missing.

**Comment by asnagni** [comment]
Hi @halx99 <https://github.com/halx99>,
   If you check on cocos2d-x projects. The Simulators and the Devices have the General tab. You can use that Genral tab to configure both of them. I’m not sure why it is missing with Axmol?  Do you have a way to activate it?

Thank you.

**Comment by halx99** [comment]
yes, the commit https://github.com/axmolengine/axmol/commit/86a2062b982e31dd02f77ad8794fc32f6bcd0ee2 remove sysroot, but break ci, and I can build for simulator by xcode

**Comment by asnagni** [comment]
Can you please reopen this issue?

Thank you

**Comment by halx99** [comment]
As rh101,  said, since axmol is a fully cmake based, any configuration and thirdparty or embed content should write cmake, so fix simulator issue is pointless

**Comment by asnagni** [comment]
The issue is not to be able to build is to have it back

**Comment by asnagni** [comment]
I understand that. But I still believe that the issue is not resolved . I’m not arguing with you, I’m trying to make the system better. That’s all, and I do believe that it is confusing and a bit challenging to migrate.
Believe me we are working together here.

Thank you,
Stay safe

**Comment by halx99** [comment]
I think that's xcode behavior, not axmol bug or cmake script bug.

**Comment by asnagni** [comment]
Hum, are you sure? So why it is working with coco2d-x. I think we missed something and maybe we could spend a bit a time looking into this. What do you think?
Like I said we are still using cocos2d-x and we are trying to migrate to Axmol and we don’t see any of that with cocos2d-x project.


Thank you,
Stay safe

**Comment by halx99** [comment]
maybe cocos2dx not set CMAKE_OSX_SYSROOT explicit

**Comment by halx99** [comment]
which cocos2dx version do you use?

**Comment by asnagni** [comment]
Thank you for reopening it, I do appreciate.

**Comment by asnagni** [comment]
We are using the versions 4.0 and 3.17.2+

I’m saying 3.17.2+ because they released unofficial fixes after the official 3.17.2

Thank you,
Stay safe

**Comment by halx99** [comment]
![image](https://github.com/axmolengine/axmol/assets/6977319/65315e17-b4b0-4001-9b9d-67ce7c6c3218)
cocos2d-x also missing `General` tab for simulator, cmake gen command is (from cocos2d-x official github):'

```
cmake -B build_ios -GXcode -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_SYSROOT=iphonesimulator
```

**Comment by asnagni** [comment]
I’m not sure you my project based on Coco2d-x . I’m not sure how you created that, but if you use the normal common that create a cpp project (iOS & Android) for you I have both:

Example:
cocos new MyGame -p com.example.mygame -l cpp -d ~/Documents/

Once it’s created just click on the Xcoded project. You don’t need any additional command.

Thank you,
Stay safe

**Comment by halx99** [comment]
I use cmake https://github.com/cocos2d/cocos2d-x/tree/v4/cmake

**Comment by halx99** [comment]
At least, you needs use cmake to generate xcodeproj, right? and the command `cocos new MyGame -p com.example.mygame -l cpp -d ~/Documents/` not generate xcodeproj for you.

**Comment by asnagni** [comment]
Ok, I see what you are doing. It looks to me that they have an issue too. I’m not sure what is the issue here. 
But don’t you think that it will make more sense from a design/architure point to have the Simulator and Device behave the same way?


Thank you,
Stay safe

**Comment by rh101** [comment]
> Hum, are you sure? So why it is working with coco2d-x. 

It feels like there is be a bit of misunderstanding here.  The only reason it is working in cocos2d-x is that it never set `CMAKE_OSX_SYSROOT`, so while you can see the "General" tab, cocos2d-x doesn't seem to have CI builds enabled, so they would have never noticed that the CI build would be broken as a result of not setting `CMAKE_OSX_SYSROOT`.

Commit https://github.com/axmolengine/axmol/commit/86a2062b982e31dd02f77ad8794fc32f6bcd0ee2 allows the Xcode simulator project to be generated with the "General" tab, so it address the issue in your original post.  The side effect of not setting `CMAKE_OSX_SYSROOT` is that the CI builds broke.

This is not an issue specific to Axmol.

**Reply by halx99** [reply, marked as answer]
The commit https://github.com/axmolengine/axmol/commit/86a2062b982e31dd02f77ad8794fc32f6bcd0ee2 cause critical issue(compiling error), so reverted: https://github.com/axmolengine/axmol/commit/36af496768a3d3cb45a7d2336477c10560a80761

**Comment by asnagni** [comment]
No that is not the way we did it and we did a long time ago now. So I don’t remember exactly. But I think you are right.

**Comment by asnagni** [comment]
Anyway we are moving actively to Axmol and fixing the issues that we are facing during the migration. I hope like in a week or so we will be over all of this 😊.

Thank you,
Stay safe

**Comment by halx99** [comment]
> > Hum, are you sure? So why it is working with coco2d-x.
> 
> It feels like there is be a bit of misunderstanding here. The only reason it is working in cocos2d-x is that it never set `CMAKE_OSX_SYSROOT`, so while you can see the "General" tab, cocos2d-x doesn't seem to have CI builds enabled, so they would have never noticed that the CI build would be broken as a result of not setting `CMAKE_OSX_SYSROOT`.
> 
> Commit [86a2062](https://github.com/axmolengine/axmol/commit/86a2062b982e31dd02f77ad8794fc32f6bcd0ee2) allows the Xcode simulator project to be generated with the "General" tab, so it address the issue in your original post. The side effect of not setting `CMAKE_OSX_SYSROOT` is that the CI builds broke.
> 
> This is not an issue specific to Axmol.

I add parameter `-sdk iphonesimulator` for ci: https://github.com/axmolengine/axmol/blob/dev/1k/build1k.ps1#L1082, should make ci works

**Comment by rh101** [comment]
@halx99 ~For CI builds, would it be possible to pass some setting to cmake when generating the project to ensure that it still does set `CMAKE_OSX_SYSROOT`?~ 

Never-mind! You answered my question just as I posted this.

**Comment by asnagni** [comment]
Hi,
   I show you the project that I have and it’s there but again it was done few years ago now so I’m not sure but I can dig into it if that can help. 

  But I still think that Axmol should aim to be better than coco2dx that is why we are migrating 😊.

Thank you,
Stay safe
